src/hotspot/share/gc/shared/taskqueue.inline.hpp

```diff
@@ -291,17 +293,19 @@ GenericTaskQueueSet<T, F>::steal_best_of_2(uint queue_num, E& t) {
     uint sz2 = _queues[k2]->size();
 
     uint sel_k = 0;
-    bool suc = false;
+    PopResult suc = T::Empty;
 
     if (sz2 > sz1) {
       sel_k = k2;
       suc = _queues[k2]->pop_global(t);
+      TASKQUEUE_STATS_ONLY(queue(queue_num)->stats.record_steal_attempt((uint)suc);)
```

Comment:
Consider adding `TaskQueueSuper::record_steal_attempt(PopResult)` to package up this idiom.